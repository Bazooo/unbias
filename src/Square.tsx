import React, { useState } from 'react'

const Square = () => {
  const [text, toggle] = useState('off')

  const onClick: React.MouseEventHandler = ({}) => {
    toggle(text === 'on' ? 'off' : 'on')
  }

  return (
    <div onClick={onClick}>{text}</div>
  )
}

export default Square
