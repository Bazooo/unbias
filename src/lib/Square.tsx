import React from 'react'

const Square = () => (
  <div>square</div>
)

export default Square
