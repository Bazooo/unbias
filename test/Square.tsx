import React from 'react'
import { shallow } from 'enzyme'
import Square from '../src/Square'

test('Square changes text on click', () => {
  const square = shallow(<Square />)

  // Interaction demo
  expect(square.text()).toEqual('off')
  square.simulate('click')
  expect(square.text()).toEqual('on')

  // Snapshot demo
  expect(square).toMatchSnapshot()
})
