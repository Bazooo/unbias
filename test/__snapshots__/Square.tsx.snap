// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square changes text on click 1`] = `
<div
  onClick={[Function]}
>
  on
</div>
`;
